### ISSUE TITLE
Invoking a default recipe implicitly.

### ISSUE THREAD

**Issue by gl-yziquel** [casey/just#2205]
Hi.

One of the reasons I like just is the ability to circumvent programming languages to implement CLIs, by using just's logic for recipes and modules instead. Which is why I wonder if the following would be a good idea:

I am currently developping a small tool called `bld` that fetches just fragments from some git repository over http(s) to build some pieces of code from sources. For instance, currently, I do `bld build buildbot` to build the buildbot python package.

I would like to do `bld buildbot` instead. However, `buildbot` is not a recipe, but rather an identifier from which I retrieve remote build instructions. So, Ideally, I'd like just to interpret `bld buildbot` as `bld default buildbot` where `default` is the default recipe.

So, basically, I'd like something like `set implicit-default-recipe` that essentially makes just behave as follow. `just dosomething` would look if dosomething is a recipe, and, if not, would attempt to perform `just default dosomething` with the default recipe.

Would that be welcome in just ? I do not believe that behaviour to currently be possible.

**Comment by casey** [comment]
I definitely see the rationale, but I think implicit behavior based on the value of an argument is probably too confusing. For example, if you were doing `just foo` which translated into `just default-recipe foo`, but then you created a recipe named `foo`, the behavior would change to executing `just foo`. I think creating a wrapper script is probably the best way to do this.
